crates/solver-service/src/apis/quote/generation.rs

```diff
@@ -688,13 +727,12 @@ impl QuoteGenerator {
 				QuoteError::InvalidRequest(format!("Invalid output chain ID: {}", e))
 			})?;
 
-			// Get settlement and selected oracle for the output chain (like permit2 flow)
+			// Get preferred settlement for the output chain (prioritizes Direct settlement like escrow)
 			let (_output_settlement, output_selected_oracle) = self
-				.settlement_service
-				.get_any_settlement_for_chain(output_chain_id)
+				.get_preferred_settlement_for_escrow(output_chain_id)
```

Comment:
I think we need to remove this function. It doesn't make sense for our purpose as it tries to prioritise 'direct' settlement.